lib/helpers/common.js

```diff
@@ -67,7 +67,19 @@ function flatten(update, path, options, schema) {
  * ignore
  */
 
-function modifiedPaths(update, path, result) {
+function modifiedPaths(update, path, result, recursion=null) {
+  if(recursion == null){
+    recursion = {
+      count: 0,
+      raw: {update, path}
+    };
+  }
+
+  recursion.count++;
+  if(recursion.count >= 1024){
+    throw new Error(`Mongoose: bad update value, ${recursion.raw.update}, ${recursion.raw.path}`);
```

Comment:
Maybe create a new MongooseError?

Reply:
I was just worried that it would cause a circular reference.

I will make a new commit tomorrow.

Reply:
Dont worry, I just think a new MongooseRecursionError or something like that would be nicer.